src/plugins/data/common/search/es_search/es_search_rxjs_utils.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { from } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import type { ShardsResponse } from 'elasticsearch';
+import type { TransportRequestPromise } from '@elastic/elasticsearch/lib/Transport';
+import type { Assign } from 'utility-types';
+
+import { ApiResponse } from '@elastic/elasticsearch';
+import { shimAbortSignal } from './shim_abort_signal';
+import { toSnakeCase } from './to_snake_case';
+import { getTotalLoaded } from './get_total_loaded';
+
+import type { IEsRawSearchResponse } from './types';
+import type { IKibanaSearchResponse } from '../types';
+
+export type SearchMethod = <SearchResponse extends IEsRawSearchResponse = IEsRawSearchResponse>(
+  params: any,
```

Comment:
Would love if we could get rid of these `any`s here... I'll think about it and see if I can offer a meaningful suggestion.